pkg/template/processor_test.go

```diff
@@ -0,0 +1,223 @@
+package template_test
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/codeready-toolchain/member-operator/pkg/template"
+
+	"github.com/go-logr/zapr"
+	apitemplate "github.com/openshift/api/template/v1"
+	"go.uber.org/zap"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	"fmt"
+	testtemplates "github.com/codeready-toolchain/member-operator/test/templates"
+	"github.com/codeready-toolchain/toolchain-common/pkg/test"
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/types"
+	"context"
+	authv1 "github.com/openshift/api/authorization/v1"
+	projectv1 "github.com/openshift/api/project/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"github.com/satori/go.uuid"
+	"github.com/codeready-toolchain/member-operator/pkg/apis"
+)
+
+func TestProcess(t *testing.T) {
```

Comment:
I'd like to also see a test with default values set inside of the templates - that they are overridden.